src/js/hooks/outboundImport/useOutboundImportForm.js

```diff
@@ -98,36 +59,51 @@ const useOutboundImportForm = ({ next }) => {
       zodResolver(validationSchema(values))(values, context, options),
   });
 
+  // TODO: implement data validation request
   // TODO: implement data validation request
   const onSubmitStockMovementDetails = (values) => {
     // here distinguish whether the onSubmit happens from detalis step or confirm page.
     // if it happens from details step, send an endpoint to validate the data,
     // if from confirm page - save & validate
-    console.log('Sending values for validation', values);
+    console.log(values.packingList);
+    const formData = new FormData();
+    formData.append('importFile', values.packingList);
+    const config = {
+      headers: {
+        'content-type': 'multipart/form-data',
+      },
+    };
+    apiClient.post('/packingList/importPackingList', formData, config).then((res) => {
+      console.log(res);
+      const basicDetails = {
+        ..._.omit(values, 'shipmentType', 'trackingNumber', 'dateShipped', 'expectedDeliveryDate', 'packingList'),
+        dateRequested: moment(values.dateRequested).format(DateFormat.MM_DD_YYYY),
+      };
+      const sendingOptions = {
+        ..._.pick(values, ['shipmentType', 'trackingNumber']),
+        expectedShippingDate: moment(values.dateShipped).format(DateFormat.MM_DD_YYYY),
+        expectedDeliveryDate: moment(values.expectedDeliveryDate).format(DateFormat.MM_DD_YYYY),
+      };
 
-    // TODO: remove this hardcoded data cases
-    //  (meant for testing only before back and front integrations)
-    if (values.description === 'ERROR') {
-      setLineItemErrors(tableErrors);
-    } else if (values.description === 'NO_PACK_1') {
-      setLineItems(otherData.map((it) => ({ ...it, palletName: undefined })));
-    } else if (values.description === 'NO_PACK_2') {
-      setLineItems(otherData.map((it) => ({ ...it, boxName: undefined })));
-    } else if (values.description === 'NO_PACK') {
-      setLineItems(otherData.map((it) => ({ ...it, boxName: undefined, palletName: undefined })));
-    } else {
-      setLineItems(otherData);
-    }
+      const items = res.data.data.map((item) => ({
+        origin: basicDetails.origin.id,
+        ...item,
+        product: item.productCode,
+        rowId: _.uniqueId(),
+      }));
 
-    /** we want to display header info only after a successful outbound upload of create step */
-    setHeaderDetailsData({
-      origin: values.origin?.name,
-      destination: values.destination?.name,
-      dateRequested: values.dateRequested,
-      description: values.description,
+      apiClient.post('/api/fulfillments/validate', {
+        fulfillmentDetails: basicDetails,
+        packingList: items,
+        sendingOptions,
+      }).then(() => apiClient.post('/api/fulfillments', {
+        fulfillmentDetails: basicDetails,
+        packingList: items,
+        sendingOptions,
+      })).then((response) => {
+        window.location = STOCK_MOVEMENT_URL.show(response.data.data.id);
+      });
```

Comment:
@awalkowiak don't look at this file at all. It's just a :spaghetti: on my testing purposes. I will make it cleaner etc in the frontend ticket.
I decided not to remove it though, to make the feature testable for Kasia and Manon